### ISSUE TITLE
Data list enhancements for better info architecture

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#821]
When testing the data list component in screen readers, VO provides a clear separation between list items, but JAWS and NVDA did not. The best suggestion I have to address this is to incorporate the use of headings into the list.

For screen reader users, heading elements are useful for indicating the beginning of a new section of content, but also have additional benefits with regard to navigation. Using heading elements will allow the user to skip through the list without reading every element in the row, so that they can more easily find the contents they're looking for.

Specifically, updates needed include:

- wherever the primary contents for a row display, replace the `<span>` with an `<h3>` element
- the styles that currently display should not be affected by using a heading element in this way
- styles should be checked to make sure the consumer could use any heading level without affecting the styles
- The accessibility section should be updated to mention the use of a heading element, and that the heading level should be defined by the consumer based on the context in which they're using the data list (e.g. if the data list is labeled with a `<h4>`, then the data list rows would use a `<h5>`)

**Comment by jgiardino** [comment]
Closing as a duplicate of #1114
